src/include/settings/settings_defs.h

```diff
@@ -185,63 +185,82 @@ SETTING_bool(
 )
 
 SETTING_bool(
-    metrics_logging,
+    use_metrics_thread,
+    "Use a thread for the metrics sub-system (default: true).",
+    true,
+    false,
+    terrier::settings::Callbacks::NoOp
+)
+
+SETTING_bool(
+    logging_metrics_enable,
     "Metrics collection for the Logging component (default: false).",
     false,
     true,
     terrier::settings::Callbacks::MetricsLogging
 )
 
 SETTING_bool(
-    metrics_transaction,
+    transaction_metrics_enable,
     "Metrics collection for the TransactionManager component (default: false).",
     false,
     true,
     terrier::settings::Callbacks::MetricsTransaction
 )
 
 SETTING_bool(
-    metrics_gc,
+    gc_metrics_enable,
     "Metrics collection for the GarbageCollector component (default: false).",
     false,
     true,
     terrier::settings::Callbacks::MetricsGC
 )
 
 SETTING_bool(
-    metrics_query_trace,
+    query_trace_metrics_enable,
     "Metrics collection for Query Traces (default: false).",
     false,
     true,
     terrier::settings::Callbacks::MetricsQueryTrace
 )
 
 SETTING_bool(
-    metrics_execution,
+    execution_metrics_enable,
     "Metrics collection for the Execution component (default: false).",
     false,
     true,
     terrier::settings::Callbacks::MetricsExecution
 )
 
 SETTING_bool(
-    metrics_pipeline,
+    pipeline_metrics_enable,
     "Metrics collection for the ExecutionEngine pipelines (default: false).",
     false,
     true,
     terrier::settings::Callbacks::MetricsPipeline
 )
 
+SETTING_int(
+    pipeline_metrics_interval,
+    "Sampling rate of metrics collection for the ExecutionEngine pipelines with 0 = 100%, 1 = 50%, "
+    "9 = 10%, X = 1/(X+1)% (default: 0 for 100%).",
+    0,
```

Comment:
I think we should keep this as 9? Let's not change this frequently without specific reason.

Reply:
Reverted to 9.